app/domain/token_factory.rb

```diff
@@ -10,9 +12,19 @@ class TokenFactory < Dry::Struct
   def signing_key(account)
     slosilo["authn:#{account}".to_sym] || raise(NoSigningKey, account)
   end
-    
+
+  def signed_host_key(account:, username:)
+    exp_t = Time.now + Rails.application.config.conjur_config.hosts_authorization_token_ttl.to_i
+    signing_key(account).issue_jwt(sub: username, exp: exp_t)
+  end
+
   def signed_token(account:, username:)
-    signing_key(account).issue_jwt(sub: username)
+    if username.starts_with?('host/')
+      signed_host_key(account: account, username: username)
+    else
+      exp_t = Time.now + Rails.application.config.conjur_config.users_authorization_token_ttl.to_i
+      signing_key(account).issue_jwt(sub: username, exp: exp_t)
+    end
```

Comment:
It looks like User and Host auth token creation is the same. What about pulling the user & host offsets into a single method?  Maybe something like:

```ruby
def signed_token(account:, username:)
  offset = if username.starts_with?('host/')
    Rails.application.config.conjur_config.hosts_authorization_token_ttl
  else
    Rails.application.config.conjur_config.users_authorization_token_ttl
  end

  signing_key(account).issue_jwt(
    sub: username, 
    exp: Time.now + offset.to_i
  )
end
```

Reply:
If we add the TTL configuration into the method arguments we can now unit test this method to target the offset.  

```ruby
def signed_token(
  account:, 
  username:, 
  host_ttl: Rails.application.config.conjur_config.host_authorization_token_ttl,
  user_ttl: Rails.application.config.conjur_config.user_authorization_token_ttl
)
  offset = username.starts_with?('host/') ? host_ttl : user_ttl

  signing_key(account).issue_jwt(
    sub: username, 
    exp: Time.now + offset.to_i
  )
end

Reply:
Please move this and other comments to be to the JIRA item - as was required by our managers

Reply:
done